File: FrontEnd-Workspace/namoo.JS.SMS/JS/Student.js
       //학생을 추상화한 클래스
        class Student{

            //static 프로퍼티
            static schoolname = "Korea IT School"

            constructor(ssn, name, korean, eng, math){
                this.ssn = ssn
                this.name = name
                this.korean = korean
                this.eng = eng
                this.math = math
            }

            set ssn(ssn){
            this._ssn = ssn;
            }
            get ssn(){
                return this._ssn
            }

            set name(name){
                if(name === undefined){
                    throw new Error("이름 미입력!");
                }
            this._name = name;
            }
            get name(){
            return this._name
            }

            set korean(korean){
            this._korean = korean;
            }
            get korean(){
            return this._korean
            }    


            set eng(eng){
            this._eng = eng;
            }
            get eng(){
            return this._eng
            }  
            
            set math(math){
            this._math = math;
            }
            get math(){
            return this._math
            }  

            getsum(){
                return this.korean + this.eng + this.math
            }
            
            getAverage(){
                return this.getsum()/3
            }
            toString(){
                return `학번: ${this.ssn},  이름: ${this.name}, 국어: ${this.korean}점/ 영어: ${this.eng}점/ 수학: ${this.math}점`
            }
        }

export{Student};

File: namoo.jdbc/src/namoo/jdbc/Student.java
package namoo.jdbc;

/**
 * OR-Mapping 클래스 또는 Domain클래스 정의
 *
 */
public class Student {
	private int ssn, korean, english, math;
	private String name;
	
	public Student() {}
	
	public Student(int ssn, int korean, int english, int math, String name) {
		this.ssn = ssn;
		this.korean = korean;
		this.english = english;
		this.math = math;
		this.name = name;
	}

	public int getSsn() {
		return ssn;
	}

	public void setSsn(int ssn) {
		this.ssn = ssn;
	}

	public int getKorean() {
		return korean;
	}

	public void setKorean(int korean) {
		this.korean = korean;
	}

	public int getEnglish() {
		return english;
	}

	public void setEnglish(int english) {
		this.english = english;
	}

	public int getMath() {
		return math;
	}

	public void setMath(int math) {
		this.math = math;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	@Override
	public String toString() {
		String str = " [Student] "+"ssn: "+ssn+" name: "+name+" korean: "+korean+" english: "+english+" math: "+math;
		return str;
		
	}
	
	
	
}
